### PR TITLE
Initial Patient and Therapist Routes

### PR DESCRIPTION
**Pull request by Hitenp1**
Created initial routes for patient and therapist -need to update depending on front end. There is also a test included for patients but has not been tested since local mocha is not working

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -2,13 +2,20 @@
...
 // routing here
 router
-  .use('/therapist', therapistRoute)
-  .use('/pathTwo', pathTwo)
+  .use('/therapist', therapist)
+  .use('/patient', patient)
+  .use('/routetemplate', routetemplate)
+
+// Send along any errors
+router.use((err, req, res, next) => {
```
Comment (jakepeyser) [inline server/routes/index.js]: We should not need this, there is a universal error handler in `app.js` already?

File: server/routes/patient.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+const planModel = db.model('plan')
+const therapistModel  = db.model('therapist') 
+const treatmentModel  = db.model('patient') 
+const workoutModel  = db.model('therapist') 
+
+// -=-=-= CREATE =-=-=-
+
+// create patient
+patientRoutes.post('/', (req, res, next) => {
+  patientModel.create(req.body)
```
Comment (jakepeyser) [inline server/routes/patient.js]: Do not pass in `req.body` directly? This is a security vulnerability, we should just map arguments by by key

File: server/routes/patient.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  .catch(next);
+});
+
+
+// -=-=-=-= UPDATE =-=-=-=-
+
+// modify patient info
+patientRoutes.put('/:id', (req, res, next) => {  
+  patientModel.findById(req.params.id)
+    .then(patient => patient.update(req.body))
```
Comment (jakepeyser) [inline server/routes/patient.js]: same thing here, we should not pass `req.body` directly in

File: server/routes/patient.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    .then(updated => res.status(201).send(updated))
+    .catch(next);
+})
+
+// -=-=-=-=-= DELETE =-=-=-=-=-
+
+// delete patient
+patientRoutes.delete('/:id', (req, res, next) => {  
+  patientModel.findById(req.params.id)
+      .then(result => result.destroy())
```
Comment (jakepeyser) [inline server/routes/patient.js]: We can just destroy using the `where` clause so you don't have to make 2 DB hits

File: server/routes/therapist.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// therapistRoutes
```
Comment (jakepeyser) [inline server/routes/therapist.js]: Similar comments to those made for the patient routes

File: tests/patient.test.js [hunk 6]
```diff
@@ -88,4 +93,45 @@ describe('Patient', function () {
       })
     })
   })
+
```
Comment (jakepeyser) [inline tests/patient.test.js]: @Hitenp1 moved your patient route tests into here to consolidate all patient tests

File: tests/patient.test.js [hunk 7]
```diff
@@ -88,4 +93,45 @@ describe('Patient', function () {
...
+  describe.only('Patient Route', () => {
+    
+    describe('Patients', () => {
+
+      it('GET all patients', function () {
+        return agent
```
Comment (jakepeyser) [inline tests/patient.test.js]: This promised needed to be returned, `done` was not being called that's why it was timing out

File: tests/patient.test.js [hunk 8]
```diff
@@ -88,4 +93,45 @@ describe('Patient', function () {
...
+        .post('/api/patient')
+        .send({
+          'first_name': 'PatientTestF1',
+          'last_name': 'PatientTestL1',
+          'email': 'patient1@test.com',
+          'DOB': '11/18/15',
+          'gender': 'F'
+        })
+        .expect(201)
+        .end(function (err, res) {
```
Comment (jakepeyser) [inline tests/patient.test.js]: Either `end` part is not working or there is something wrong with the route. I recommend testing the route with Postman/cUrl to eliminate that possibility

**Comment by Hitenp1** [comment]
Still have outstanding fixes from the comments , specifically the routes for update(removing req.body) and the .end not working on the patient tests

**Pull request merged**